### ISSUE TITLE
Ext-dns crashing when "sources" are being used

### ISSUE THREAD

**Issue by EKami** [argoflow/argoflow-aws#169]
I'm getting this erro in `ext-dns`:
```
$ k logs external-dns-5bd854dd6-ljgpz -n kube-system --previous
Enter passphrase to unlock /home/ekami/.awsvault/keys/:
time="2021-07-11T13:18:22Z" level=info msg="config: {APIServerURL: KubeConfig: RequestTimeout:30s ContourLoadBalancerService:heptio-contour/contour GlooNamespace:gloo-system SkipperRouteGroupVersion:zalando.org/v1 Sources:[service ingress istio-gateway istio-virtualservice] Namespace: AnnotationFilter: LabelFilter: FQDNTemplate: CombineFQDNAndAnnotation:false IgnoreHostnameAnnotation:false IgnoreIngressTLSSpec:false Compatibility: PublishInternal:false PublishHostIP:false AlwaysPublishNotReadyAddresses:false ConnectorSourceServer:localhost:8080 Provider:aws GoogleProject: GoogleBatchChangeSize:1000 GoogleBatchChangeInterval:1s DomainFilter:[pathwatch-research.com] ExcludeDomains:[] RegexDomainFilter: RegexDomainExclusion: ZoneNameFilter:[] ZoneIDFilter:[] AlibabaCloudConfigFile:/etc/kubernetes/alibaba-cloud.json AlibabaCloudZoneType: AWSZoneType: AWSZoneTagFilter:[] AWSAssumeRole: AWSBatchChangeSize:1000 AWSBatchChangeInterval:1s AWSEvaluateTargetHealth:true AWSAPIRetries:3 AWSPreferCNAME:false AWSZoneCacheDuration:0s AzureConfigFile:/etc/kubernetes/azure.json AzureResourceGroup: AzureSubscriptionID: AzureUserAssignedIdentityClientID: BluecatConfigFile:/etc/kubernetes/bluecat.json CloudflareProxied:false CloudflareZonesPerPage:50 CoreDNSPrefix:/skydns/ RcodezeroTXTEncrypt:false AkamaiServiceConsumerDomain: AkamaiClientToken: AkamaiClientSecret: AkamaiAccessToken: AkamaiEdgercPath: AkamaiEdgercSection: InfobloxGridHost: InfobloxWapiPort:443 InfobloxWapiUsername:admin InfobloxWapiPassword: InfobloxWapiVersion:2.3.1 InfobloxSSLVerify:true InfobloxView: InfobloxMaxResults:0 DynCustomerName: DynUsername: DynPassword: DynMinTTLSeconds:0 OCIConfigFile:/etc/kubernetes/oci.yaml InMemoryZones:[] OVHEndpoint:ovh-eu OVHApiRateLimit:20 PDNSServer:http://localhost:8081 PDNSAPIKey: PDNSTLSEnabled:false TLSCA: TLSClientCert: TLSClientCertKey: Policy:upsert-only Registry:txt TXTOwnerID:default TXTPrefix: TXTSuffix: Interval:1m0s MinEventSyncInterval:5s Once:false DryRun:false UpdateEvents:false LogFormat:text MetricsAddress::7979 LogLevel:info TXTCacheInterval:0s TXTWildcardReplacement: ExoscaleEndpoint:https://api.exoscale.ch/dns ExoscaleAPIKey: ExoscaleAPISecret: CRDSourceAPIVersion:externaldns.k8s.io/v1alpha1 CRDSourceKind:DNSEndpoint ServiceTypeFilter:[] CFAPIEndpoint: CFUsername: CFPassword: RFC2136Host: RFC2136Port:0 RFC2136Zone: RFC2136Insecure:false RFC2136GSSTSIG:false RFC2136KerberosRealm: RFC2136KerberosUsername: RFC2136KerberosPassword: RFC2136TSIGKeyName: RFC2136TSIGSecret: RFC2136TSIGSecretAlg: RFC2136TAXFR:false RFC2136MinTTL:0s NS1Endpoint: NS1IgnoreSSL:false NS1MinTTLSeconds:0 TransIPAccountName: TransIPPrivateKeyFile: DigitalOceanAPIPageSize:50 ManagedDNSRecordTypes:[A CNAME] GoDaddyAPIKey: GoDaddySecretKey: GoDaddyTTL:0 GoDaddyOTE:false}"
time="2021-07-11T13:18:22Z" level=info msg="Instantiating new Kubernetes client"
time="2021-07-11T13:18:22Z" level=info msg="Using inCluster-config based on serviceaccount-token"
time="2021-07-11T13:18:22Z" level=info msg="Created Kubernetes client https://172.20.0.1:443"
time="2021-07-11T13:19:25Z" level=fatal msg="failed to sync cache: timed out waiting for the condition"
```
Resulting in a `CrashLoopBackOff` of the pod.
However if I remove these lines from [this file](https://github.com/argoflow/argoflow-aws/blob/master/distribution/argocd-applications/external-dns.yaml):
```
      - name: sources[0]
        value: service
      - name: sources[1]
        value: ingress
      - name: sources[2]
        value: istio-gateway
      - name: sources[3]
        value: istio-virtualservice
```

Then the pod gets healthy and stops crashing.
Any idea what's going on?

Thanks!

**Comment by DavidSpek** [comment]
@EKami Removing those lines essentially disables all the external DNS functionalities. What version of the chart are you using? Have you already checked the upstream project for similar issues?

**Comment by EKami** [comment]
Hi David, I'm using the exact same configuration as the one shared in this repo, I didn't change anything. This is from my repo `external-dns.yaml`:
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: external-dns
  namespace: argocd
spec:
  destination:
    namespace: kube-system
    server: https://kubernetes.default.svc
  project: default
  source:
    chart: external-dns
    helm:
      parameters:
      - name: aws.region
        value: <<__aws_region__>>
      - name: domainFilters[0]
        value: <<__route53.hosted_zone_domain__>>
      - name: serviceAccount.create
        value: 'true'
      - name: rbac.create
        value: 'true'
      - name: resources.limits.cpu
        value: 100m
      - name: resources.limits.memory
        value: 300Mi
      - name: resources.requests.cpu
        value: 100m
      - name: resources.requests.memory
        value: 300Mi
      - name: sources[0]
        value: service
      - name: sources[1]
        value: ingress
      - name: sources[2]
        value: istio-gateway
      - name: sources[3]
        value: istio-virtualservice
      # - name: aws.assumeRolenArn
      #   value: <<__role_arn.external_dns__>>
      - name: serviceAccount.annotations.eks\.amazonaws\.com/role-arn
        value: <<__role_arn.external_dns__>>
    repoURL: https://charts.bitnami.com/bitnami
    targetRevision: 5.1.1
  syncPolicy:
    automated:
      prune: true
      selfHeal: true
 ```

**Comment by DavidSpek** [comment]
Were the values properly replaced before you deployed external DNS? I still see the placeholders in the yaml file you shared.

**Comment by EKami** [comment]
And the terraform config to create the role:
```
resource "aws_iam_policy" "ext_dns" {
  name_prefix = "eks-ext-dns-${module.eks.cluster_id}"
  description = "EKS ext-dns for cluster ${module.eks.cluster_id}"
  policy = data.aws_iam_policy_document.ext_dns.json
}

data "aws_iam_policy_document" "ext_dns" {
  statement {
    effect = "Allow"

    actions = [
      "route53:GetChange"
    ]

    resources = [
      "arn:aws:route53:::change/*"]
  }

  statement {
    effect = "Allow"

    actions = [
      "route53:ChangeResourceRecordSets",
      "route53:ListResourceRecordSets"
    ]

    resources = [
      "arn:aws:route53:::hostedzone/*"]
  }

  statement {
    effect = "Allow"

    actions = [
      "route53:ListHostedZones",
      "route53:ListResourceRecordSets"
    ]

    resources = ["*"]
  }
}


resource "aws_acm_certificate" "cert" {
  domain_name = "${var.domain_name}.${var.hosted_dns_name}"
  validation_method = "DNS"

  lifecycle {
    create_before_destroy = true
  }
}

resource "aws_acm_certificate_validation" "cert_val" {
  certificate_arn = aws_acm_certificate.cert.arn
  validation_record_fqdns = [for record in aws_route53_record.ext-dns-route : record.fqdn]
}

data "aws_route53_zone" "domain" {
  name = var.hosted_dns_name
  private_zone = false
}

resource "aws_route53_record" "ext-dns-route" {
  for_each = {
    for dvo in aws_acm_certificate.cert.domain_validation_options : dvo.domain_name => {
      name   = dvo.resource_record_name
      record = dvo.resource_record_value
      type   = dvo.resource_record_type
    }
  }

  allow_overwrite = true
  name            = each.value.name
  records         = [each.value.record]
  ttl             = 60
  type            = each.value.type
  zone_id         = data.aws_route53_zone.domain.zone_id
}
```
with:
```
output "argoflow_config" {
  value = <<CONFIG
<<__role_arn.external_dns__>>=${module.iam_roles_for_services_accounts_with_oidc.iam_role_arn}
}
```

**Comment by EKami** [comment]
> Were the values properly replaced before you deployed external DNS? I still see the placeholders in the yaml file you shared, if you deployed that file it would indeed fail.

Yes, here is what the replacement looks like:
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: external-dns
  namespace: argocd
spec:
  destination:
    namespace: kube-system
    server: https://kubernetes.default.svc
  project: default
  source:
    chart: external-dns
    helm:
      parameters:
      - name: aws.region
        value: us-east-2
      - name: domainFilters[0]
        value: pathwatch-research.com
      - name: serviceAccount.create
        value: 'true'
      - name: rbac.create
        value: 'true'
      - name: resources.limits.cpu
        value: 100m
      - name: resources.limits.memory
        value: 300Mi
      - name: resources.requests.cpu
        value: 100m
      - name: resources.requests.memory
        value: 300Mi
      - name: sources[0]
        value: service
      - name: sources[1]
        value: ingress
      - name: sources[2]
        value: istio-gateway
      - name: sources[3]
        value: istio-virtualservice
      # - name: aws.assumeRolenArn
      #   value: arn:aws:iam::122803953293:role/eks-cluster-kubeflow-test-irsa
      - name: serviceAccount.annotations.eks\.amazonaws\.com/role-arn
        value: arn:aws:iam::122803953293:role/eks-cluster-kubeflow-test-irsa
    repoURL: https://charts.bitnami.com/bitnami
    targetRevision: 5.1.1
  syncPolicy:
    automated:
      prune: true
      selfHeal: true
```

**Comment by DavidSpek** [comment]
I'm guessing this is most likely caused by incorrect IAM permissions. Have you looked at https://github.com/kubernetes-sigs/external-dns/issues/1840 already?

**Comment by EKami** [comment]
They are mentionning:
> You also need endpoints as listed in the official manifests: https://github.com/kubernetes-sigs/external-dns/blob/master/kustomize/external-dns-clusterrole.yaml .

But I have no idea where to add those endpoints

**Comment by EKami** [comment]
As for the IAM permissions they look correct to me :thinking:

**Comment by EKami** [comment]
Adding the `ClusterRole` didn't change anything, the pod is still crashing :(

**Comment by EKami** [comment]
So this is what my ClusterRole looks like with:
```
      - name: rbac.create
        value: 'true'
```
I get:
```
$ k get ClusterRoleBinding/external-dns -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.authorization.k8s.io/v1","kind":"ClusterRoleBinding","metadata":{"annotations":{},"labels":{"app.kubernetes.io/instance":"external-dns","app.kubernetes.io/managed-by":"Helm","app.kubernetes.io/name":"external-dns","helm.sh/chart":"external-dns-5.1.1"},"name":"external-dns"},"roleRef":{"apiGroup":"rbac.authorization.k8s.io","kind":"ClusterRole","name":"external-dns"},"subjects":[{"kind":"ServiceAccount","name":"external-dns","namespace":"kube-system"}]}
  creationTimestamp: "2021-07-17T12:09:42Z"
  labels:
    app.kubernetes.io/instance: external-dns
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: external-dns
    helm.sh/chart: external-dns-5.1.1
  managedFields:
  - apiVersion: rbac.authorization.k8s.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations: {}
        f:labels:
          .: {}
          f:app.kubernetes.io/instance: {}
      f:roleRef:
        f:apiGroup: {}
        f:kind: {}
        f:name: {}
      f:subjects: {}
    manager: kubectl
    operation: Update
    time: "2021-07-17T12:09:42Z"
  - apiVersion: rbac.authorization.k8s.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:kubectl.kubernetes.io/last-applied-configuration: {}
        f:labels:
          f:app.kubernetes.io/managed-by: {}
          f:app.kubernetes.io/name: {}
          f:helm.sh/chart: {}
    manager: argocd-application-controller
    operation: Update
    time: "2021-07-17T12:14:31Z"
  name: external-dns
  resourceVersion: "11850"
  uid: a807fdc9-a80d-4b76-ac73-b222b9ff339b
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: external-dns
subjects:
- kind: ServiceAccount
  name: external-dns
  namespace: kube-system
```
and:
```
$ k get ClusterRole/external-dns -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.authorization.k8s.io/v1","kind":"ClusterRole","metadata":{"annotations":{},"labels":{"app.kubernetes.io/instance":"external-dns","app.kubernetes.io/managed-by":"Helm","app.kubernetes.io/name":"external-dns","helm.sh/chart":"external-dns-5.1.1"},"name":"external-dns"},"rules":[{"apiGroups":[""],"resources":["services","pods","nodes","endpoints"],"verbs":["get","list","watch"]},{"apiGroups":["extensions","networking.k8s.io","getambassador.io"],"resources":["ingresses","hosts"],"verbs":["get","list","watch"]},{"apiGroups":["route.openshift.io"],"resources":["routes"],"verbs":["get","list","watch"]},{"apiGroups":["networking.istio.io"],"resources":["gateways","virtualservices"],"verbs":["get","list","watch"]},{"apiGroups":["zalando.org"],"resources":["routegroups"],"verbs":["get","list","watch"]},{"apiGroups":["zalando.org"],"resources":["routegroups/status"],"verbs":["patch","update"]},{"apiGroups":["projectcontour.io"],"resources":["httpproxies"],"verbs":["get","watch","list"]},{"apiGroups":["gloo.solo.io","gateway.solo.io"],"resources":["proxies","virtualservices"],"verbs":["get","list","watch"]}]}
  creationTimestamp: "2021-07-17T12:09:41Z"
  labels:
    app.kubernetes.io/instance: external-dns
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: external-dns
    helm.sh/chart: external-dns-5.1.1
  managedFields:
  - apiVersion: rbac.authorization.k8s.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations: {}
    manager: kubectl
    operation: Update
    time: "2021-07-17T12:09:41Z"
  - apiVersion: rbac.authorization.k8s.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:kubectl.kubernetes.io/last-applied-configuration: {}
        f:labels:
          .: {}
          f:app.kubernetes.io/instance: {}
          f:app.kubernetes.io/managed-by: {}
          f:app.kubernetes.io/name: {}
          f:helm.sh/chart: {}
      f:rules: {}
    manager: argocd-application-controller
    operation: Update
    time: "2021-07-17T12:14:25Z"
  name: external-dns
  resourceVersion: "11768"
  uid: fbc75d4e-e905-4b31-96a4-b97ff1d4d660
rules:
- apiGroups:
  - ""
  resources:
  - services
  - pods
  - nodes
  - endpoints
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - extensions
  - networking.k8s.io
  - getambassador.io
  resources:
  - ingresses
  - hosts
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - route.openshift.io
  resources:
  - routes
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - networking.istio.io
  resources:
  - gateways
  - virtualservices
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - zalando.org
  resources:
  - routegroups
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - zalando.org
  resources:
  - routegroups/status
  verbs:
  - patch
  - update
- apiGroups:
  - projectcontour.io
  resources:
  - httpproxies
  verbs:
  - get
  - watch
  - list
- apiGroups:
  - gloo.solo.io
  - gateway.solo.io
  resources:
  - proxies
  - virtualservices
  verbs:
  - get
  - list
  - watch
```

I have no idea why `projectcontour.io`, `gloo.solo.io`, `gateway.solo.io` and `zalando.org` are here, this looks very fishy to me to have these as default. But anyway, all the authorizations on the page you shared @DavidSpek seem to be already present.
I'll try to use my own custom-defined RBAC instead.

**Comment by EKami** [comment]
Using:
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: external-dns
rules:
  - apiGroups: ['']
    resources: ['endpoints', 'pods', 'services']
    verbs: ['get', 'watch', 'list']
  - apiGroups: ['extensions']
    resources: ['ingresses']
    verbs: ['get', 'watch', 'list']
  - apiGroups: ['']
    resources: ['nodes']
    verbs: ['get', 'list', 'watch']
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  labels:
    app.kubernetes.io/instance: external-dns
  name: external-dns
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: external-dns
subjects:
- kind: ServiceAccount
  name: external-dns
  namespace: kube-system
---
# https://artifacthub.io/packages/helm/bitnami/external-dns
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: external-dns
  namespace: argocd
spec:
  destination:
    namespace: kube-system
    server: https://kubernetes.default.svc
  project: default
  source:
    chart: external-dns
    helm:
      parameters:
      - name: aws.region
        value: <<__aws_region__>>
      - name: domainFilters[0]
        value: <<__route53.hosted_zone_domain__>>
      - name: serviceAccount.create
        value: 'true'
      - name: rbac.create
        value: 'false'
      - name: resources.limits.cpu
        value: 100m
      - name: resources.limits.memory
        value: 300Mi
      - name: resources.requests.cpu
        value: 100m
      - name: resources.requests.memory
        value: 300Mi
      - name: sources[0]
        value: service
      - name: sources[1]
        value: ingress
      - name: sources[2]
        value: istio-gateway
      - name: sources[3]
        value: istio-virtualservice
      # - name: aws.assumeRolenArn
      #   value: <<__role_arn.external_dns__>>
      - name: serviceAccount.annotations.eks\.amazonaws\.com/role-arn
        value: <<__role_arn.external_dns__>>
    repoURL: https://charts.bitnami.com/bitnami
    targetRevision: 5.1.1
  syncPolicy:
    automated:
      prune: true
      selfHeal: true
```
Gives the same result :(

**Comment by EKami** [comment]
Update: So it seems if I'm able to get past [this issue](https://github.com/argoflow/argoflow-aws/issues/176) then the error gets a bit more explicit:
```
k logs pod/external-dns-5bd854dd6-26rmt -n kube-system
Enter passphrase to unlock /home/ekami/.awsvault/keys/:
time="2021-07-17T22:15:27Z" level=info msg="config: {APIServerURL: KubeConfig: RequestTimeout:30s ContourLoadBalancerService:heptio-contour/contour GlooNamespace:gloo-system SkipperRouteGroupVersion:zalando.org/v1 Sources:[service ingress istio-gateway istio-virtualservice] Namespace: AnnotationFilter: LabelFilter: FQDNTemplate: CombineFQDNAndAnnotation:false IgnoreHostnameAnnotation:false IgnoreIngressTLSSpec:false Compatibility: PublishInternal:false PublishHostIP:false AlwaysPublishNotReadyAddresses:false ConnectorSourceServer:localhost:8080 Provider:aws GoogleProject: GoogleBatchChangeSize:1000 GoogleBatchChangeInterval:1s DomainFilter:[pathwatch-research.com] ExcludeDomains:[] RegexDomainFilter: RegexDomainExclusion: ZoneNameFilter:[] ZoneIDFilter:[] AlibabaCloudConfigFile:/etc/kubernetes/alibaba-cloud.json AlibabaCloudZoneType: AWSZoneType: AWSZoneTagFilter:[] AWSAssumeRole: AWSBatchChangeSize:1000 AWSBatchChangeInterval:1s AWSEvaluateTargetHealth:true AWSAPIRetries:3 AWSPreferCNAME:false AWSZoneCacheDuration:0s AzureConfigFile:/etc/kubernetes/azure.json AzureResourceGroup: AzureSubscriptionID: AzureUserAssignedIdentityClientID: BluecatConfigFile:/etc/kubernetes/bluecat.json CloudflareProxied:false CloudflareZonesPerPage:50 CoreDNSPrefix:/skydns/ RcodezeroTXTEncrypt:false AkamaiServiceConsumerDomain: AkamaiClientToken: AkamaiClientSecret: AkamaiAccessToken: AkamaiEdgercPath: AkamaiEdgercSection: InfobloxGridHost: InfobloxWapiPort:443 InfobloxWapiUsername:admin InfobloxWapiPassword: InfobloxWapiVersion:2.3.1 InfobloxSSLVerify:true InfobloxView: InfobloxMaxResults:0 DynCustomerName: DynUsername: DynPassword: DynMinTTLSeconds:0 OCIConfigFile:/etc/kubernetes/oci.yaml InMemoryZones:[] OVHEndpoint:ovh-eu OVHApiRateLimit:20 PDNSServer:http://localhost:8081 PDNSAPIKey: PDNSTLSEnabled:false TLSCA: TLSClientCert: TLSClientCertKey: Policy:upsert-only Registry:txt TXTOwnerID:default TXTPrefix: TXTSuffix: Interval:1m0s MinEventSyncInterval:5s Once:false DryRun:false UpdateEvents:false LogFormat:text MetricsAddress::7979 LogLevel:info TXTCacheInterval:0s TXTWildcardReplacement: ExoscaleEndpoint:https://api.exoscale.ch/dns ExoscaleAPIKey: ExoscaleAPISecret: CRDSourceAPIVersion:externaldns.k8s.io/v1alpha1 CRDSourceKind:DNSEndpoint ServiceTypeFilter:[] CFAPIEndpoint: CFUsername: CFPassword: RFC2136Host: RFC2136Port:0 RFC2136Zone: RFC2136Insecure:false RFC2136GSSTSIG:false RFC2136KerberosRealm: RFC2136KerberosUsername: RFC2136KerberosPassword: RFC2136TSIGKeyName: RFC2136TSIGSecret: RFC2136TSIGSecretAlg: RFC2136TAXFR:false RFC2136MinTTL:0s NS1Endpoint: NS1IgnoreSSL:false NS1MinTTLSeconds:0 TransIPAccountName: TransIPPrivateKeyFile: DigitalOceanAPIPageSize:50 ManagedDNSRecordTypes:[A CNAME] GoDaddyAPIKey: GoDaddySecretKey: GoDaddyTTL:0 GoDaddyOTE:false}"
time="2021-07-17T22:15:27Z" level=info msg="Instantiating new Kubernetes client"
time="2021-07-17T22:15:27Z" level=info msg="Using inCluster-config based on serviceaccount-token"
time="2021-07-17T22:15:27Z" level=info msg="Created Kubernetes client https://172.20.0.1:443"
time="2021-07-17T22:15:39Z" level=error msg="records retrieval failed: failed to list hosted zones: AccessDenied: User: arn:aws:sts::197803953293:assumed-role/eks-cluster-kubeflow-test-irsa/1626560138802632220 is not authorized to perform: route53:ListHostedZones\n\tstatus code: 403, request id: a82455a5-fd57-49bf-941f-943ada427ab2"
time="2021-07-17T22:16:39Z" level=error msg="records retrieval failed: failed to list hosted zones: AccessDenied: User: arn:aws:sts::197803953293:assumed-role/eks-cluster-kubeflow-test-irsa/1626560138802632220 is not authorized to perform: route53:ListHostedZones\n\tstatus code: 403, request id: 9dae4579-c91e-434a-adf5-4e278cce2f63"
```

This would make sense as what made the pod crash in the first place were those two lines:
```
      - name: sources[2]
        value: istio-gateway
      - name: sources[3]
        value: istio-virtualservice
```

which could not work if `istio-operator` did not properly deploy the crds, thus I really need to find a way to fix #176 first

**Comment by EKami** [comment]
Solved by solving issue #176 , sorry for the trouble.
